### PR TITLE
Add draft post about lenses & multiple inheritance

### PR DESCRIPTION
**Pull request by benjamin-hodgson**

### CODE REVIEW

File: drafts/2018-11-30-zooming-in-on-default-interface-methods.md [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+
+**PICTURE**
+
+So a lens picks out a single property inside a given object.
+
+The power of lenses comes from their composability. Given a lens identifying a `T2` inside a `T1` (`ILens<T1, T2>`) and a lens identifying a `T3` inside a `T2` (`ILens<T2, T3>`), you can compose those lenses together to focus all the way from the `T1` to the `T3`.
+
+**PICTURE**
+
+You can traverse any relationship in your data model by composing together a small number of atomic lenses. Composing lenses is so important that I've given it the shortest name I can think of: `_`. (Readers of [an eariler post of mine](2018-03-16-eighty.html) will know of my fondness for `_`.)
```
Comment (laura-campbell) [inline drafts/2018-11-30-zooming-in-on-default-interface-methods.md]: what is an atomic lens?

File: drafts/2018-11-30-zooming-in-on-default-interface-methods.md [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+
+        public T3 Get(T1 obj) => _l2.Get(_l1.Get(obj));
+
+        public T1 Set(T1 oldObj, T3 newVal)
+            => _l1.Set(oldObj, _l2.Set(_l1.Get(oldObj), newVal));
+    }
+}
+```
+
+To summarise, a lens is a way to focus on a small part of a big immutable structure. They're like a first-class version of the `.` and `=` operators: you can compose lenses together to focus deeper, and upon focusing on a location you can get and set the value at that location.
```
Comment (laura-campbell) [inline drafts/2018-11-30-zooming-in-on-default-interface-methods.md:136]: A+ I ACTUALLY UNDERSTAND THIS WOOOOO!

**Review by laura-campbell — COMMENTED** [review]
this one was easier for me to grasp! hoorah!

File: drafts/2018-11-30-zooming-in-on-default-interface-methods.md (hunk 1 above)
Comment (benjamin-hodgson) [reply drafts/2018-11-30-zooming-in-on-default-interface-methods.md]: Good question! I'm using "atomic" in the usual English (slash Greek) sense of "indivisible". Do you think I should rephrase?

**Comment by SaraJo** [comment]
This is a great post, I especially loved the business examples listed.

**Pull request merged**